### PR TITLE
Create profile_plot_from_erddap.py

### PR DESCRIPTION
**Pull request by leilabbb**
this is a modified version of generate_profile_plot.py
the main change was to fetch data from ERDDAP rather than THREDDS

### CODE REVIEW

**Comment by Bobfrat** [comment]
Instead of a new file, can you update `aws/docker/worker/generate_profile_plot.py` instead?

File: aws/docker/worker/profile_plot_from_erddap.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+# In[ ]:
+
+
+dataset_id = 'ce_311-20200708T1723'
+
+#'ce_381-20171208T1841-delayed'
+#
+#'bios_minnie-20170616T1454'
+#'amelia-20180501T0000'
+#whoi_406-20160902T1700'
```
Comment (Bobfrat) [inline aws/docker/worker/profile_plot_from_erddap.py:34]: Can you remove any extraneous comments, jupyter notebook artifacts, and debugging vars?

File: aws/docker/worker/profile_plot_from_erddap.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+import io
+import logging
+import matplotlib.cm as cm
+import matplotlib.pyplot as plt
+import netCDF4
+import numpy as np
+import traceback
+from datetime import datetime
+
+from erddapy import ERDDAP
```
Comment (Bobfrat) [inline aws/docker/worker/profile_plot_from_erddap.py:22]: We'll need to add any new dependencies to https://github.com/ioos/glider-dac-status/blob/master/aws/docker/worker/requirements.txt

Comment (leilabbb) [reply aws/docker/worker/profile_plot_from_erddap.py:22]: i will check the dependencies, rename the file, and remove  extraneous comments from the script

**Comment by Bobfrat** [comment]
@leilabbb you'll need to rebase to get the most recent changes and resolve conflicts. Also, please remove the deprecated folder, we don't need to keep the old code around.

File: aws/docker/worker/generate_profile_plot.py [hunk 3]
```diff
@@ -33,64 +33,122 @@
...
+    e.dataset_id = dataset_id
+    e.variables = [
+                    'depth',
+                    'latitude',
+                    'longitude',
+                    'salinity',
+                    'temperature',
+                    'conductivity',
+                    'density',
+                    'time',
```
Comment (Bobfrat) [inline aws/docker/worker/generate_profile_plot.py:75]: Moving forward we're going to want to plot more variables such as oxygen, pH, CDOM, etc. Those variables dont always have the same variable name. We will likely need a function that generates this list of variables based on what's available in the dataset (using standard names).

File: aws/docker/worker/profile_plot_from_erddap.py (hunk 1 above)
Comment (leilabbb) [reply aws/docker/worker/profile_plot_from_erddap.py]: done

File: aws/docker/worker/profile_plot_from_erddap.py (hunk 2 above)
Comment (leilabbb) [reply aws/docker/worker/profile_plot_from_erddap.py]: done

File: aws/docker/worker/generate_profile_plot.py [hunk 4]
```diff
@@ -35,105 +37,167 @@
...
+
+    # select only non-nan values
+    yv = [y[n] for n in ii]
+    xv = [x[n] for n in ii]
+    zv = [z[n] for n in ii]
+
+    # convert arrays to datatime or float to mask invalid values in the next step
+    xv = np.array(xv, dtype='datetime64')
+    yv = np.array(yv, dtype='float')
+    zv = np.array(zv, dtype='float')
```
Comment (Bobfrat) [inline aws/docker/worker/generate_profile_plot.py:128]: Is there a more efficient way to do this? If I'm following correctly, you're creating numpy arrays for x, y z, where there are non Nan time indices? 

Can you do something like this:

```
    # get the data arrays
    y = dataset[y_name[0]].values
    z = dataset[z_name[0]].values
    x = dataset[x_name[0]].values
   
    xv = x[~np.isnan(x)]
    yv = y[~np.isnan(x)]
    zv = z[~np.isnan(x)]
```
From there you can cast/convert to the right type (datetime/floats).

Comment (Bobfrat) [reply aws/docker/worker/generate_profile_plot.py:128]: @leilabbb for the sake of timeliness in getting this update in, I'll merge and we can handle efficiency issues (and new variables) in a separate PR

**Pull request merged**